api/v1alpha4/azurecluster_default.go

```diff
@@ -202,18 +202,18 @@ func (c *AzureCluster) setNodeOutboundLBDefaults() {
 			return
 		}
 
-		var oneSubnetWithoutNatGateway bool
+		var needsOutboundLB bool
```

Comment:
this change aims to make the intent of this bool flag a little more semantic (though the current one is literally correct, it's not as obvious what we're aiming for)